base_optional_quick_create/model.py

```diff
@@ -38,16 +39,42 @@ def wrapper(cr, uid, name, context=None):
         return wrapper
 
     def _register_hook(self, cr, ids=None):
+        """
+        On any model that has been configured to disable quick creation,
+        replace the name_create method by a method that raises an exception,
+        as this will trigger the full form creation popup in the client,
+        effectively disabling the quick create method.
+
+        Without the ids argument, we check all models, but not if we
+        have seen them before (marked by the check_quick_create attribute).
+        """
+        logger = logging.getLogger(
+            'openerp.addons.base_optional_quick_create')
+        force = True
         if ids is None:
             ids = self.search(cr, SUPERUSER_ID, [])
+            force = False
         for model in self.browse(cr, SUPERUSER_ID, ids):
+            model_obj = self.pool.get(model.model)
+            if not model_obj or (not force and hasattr(
+                    model_obj, 'check_quick_create')):
+                continue
             if model.avoid_quick_create:
-                model_name = model.model
-                model_obj = self.pool.get(model_name)
-                if model_obj and not hasattr(model_obj, 'check_quick_create'):
+                if not hasattr(model_obj, 'name_create_orig'):
+                    logger.debug(
+                        'Replacing original name_create method on %s',
+                        model.model)
+                    model_obj.name_create_orig = model_obj.name_create
                     model_obj.name_create = self._wrap_name_create(
-                        model_obj.name_create, model_name)
-                    model_obj.check_quick_create = True
+                        model_obj.name_create, model.name)
+            else:
+                if hasattr(model_obj, 'name_create_orig'):
+                    logger.debug(
+                        'Restoring original name_create method on %s',
+                        model.model)
+                    model_obj.name_create = model_obj.name_create_orig
```

Comment:
if you take my comment above, you can use `_revert_method` here: https://github.com/OCA/OCB/blob/9.0/openerp/models.py#L5299